File: tmp_six_lang_depth_repo/javascript/service.js
class BaseError extends Error {}

class ServiceError extends BaseError {}

class BaseService {
  run(value) {
    return value;
  }
}

class UserService extends BaseService {
  constructor() {
    super();
    this.count = 0;
  }

  validate(value) {
    this.count = this.count + 1;
    if (value == null) {
      throw new ServiceError("missing");
    }
    return this.run(value);
  }
}

export function entry(value) {
  const service = new UserService();
  return service.validate(value);
}

File: tmp_six_lang_depth_repo/python/service.py
class BaseError(Exception):
    pass


class ServiceError(BaseError):
    pass


class BaseService:
    def run(self, value):
        return value


class UserService(BaseService):
    def __init__(self):
        self.count = 0

    def validate(self, value):
        self.count = self.count + 1
        if value is None:
            raise ServiceError("missing")
        return self.run(value)


def entry(value):
    service = UserService()
    return service.validate(value)
